internal/encryption/encryptor.go

```diff
@@ -42,21 +45,23 @@ func generateEncryptionKey(_, _ string) ([]byte, error) {
 // generateTestEncryptionKey generates a deterministic encryption key for testing.
 // While testing, we also want to make sure different keys are generated for different docs and fields
 // and that's why we use the docID and fieldName to generate the key.
-func generateTestEncryptionKey(docID, fieldName string) ([]byte, error) {
-	return []byte(fieldName + docID + testEncryptionKey)[0:keyLength], nil
+func generateTestEncryptionKey(docID string, fieldName immutable.Option[string]) ([]byte, error) {
+	return []byte(fieldName.Value() + docID + testEncryptionKey)[0:keyLength], nil
 }
 
 // DocEncryptor is a document encryptor that encrypts and decrypts individual document fields.
 // It acts based on the configuration [DocEncConfig] provided and data stored in the provided store.
 // It uses [core.EncStoreDocKey] to store and retrieve encryption keys.
 type DocEncryptor struct {
-	conf  immutable.Option[DocEncConfig]
-	ctx   context.Context
-	store datastore.DSReaderWriter
+	conf          immutable.Option[DocEncConfig]
+	ctx           context.Context
+	store         datastore.DSReaderWriter
+	cache         map[core.EncStoreDocKey][]byte
+	generatedKeys []core.EncStoreDocKey
```

Comment:
question: Was this helping you solve the "chicken and egg" problem you described? 

The problem here is that similar to the cache, this is an infinite growing list. I also don't think its necessary. I believe your issue is how you are trying to resolve the key for a given document. However, you have already stored the key in the `EncStore` with a DocID and FieldID prefix, you should be able to query that store for the generated keys.